sys/kern/vmem.c

```diff
@@ -242,24 +253,30 @@ int vmem_add(vmem_t *vm, vmem_addr_t addr, vmem_size_t size) {
   btfree->bt_start = addr;
   btfree->bt_size = size;
 
-  WITH_MTX_LOCK (&vm->vm_lock) {
-    bt_insseg_tail(vm, btspan);
-    bt_insseg_after(vm, btfree, btspan);
-    bt_insfree(vm, btfree);
-    vm->vm_size += size;
-    vmem_check_sanity(vm);
-  }
+  bt_insseg_tail(vm, btspan);
+  bt_insseg_after(vm, btfree, btspan);
+  bt_insfree(vm, btfree);
+  vm->vm_size += size;
+  vmem_check_sanity(vm);
 
   klog("%s: added [%p-%p] span to '%s'", __func__, addr, addr + size - 1,
        vm->vm_name);
 
   return 0;
 }
 
+int vmem_add(vmem_t *vm, vmem_addr_t addr, vmem_size_t size) {
+  int error;
+  WITH_MTX_LOCK (&vm->vm_lock)
+    error = vmem_add_nolock(vm, addr, size);
+  return error;
```

Comment:
Why not:
```c
SCOPED_MTX_LOCK(&vm->vm_lock);
return vmem_add_nolock(vm, addr, size);
```

Reply:
No reason for that. I changed it according to your suggestion.